### PR TITLE
update product release number entities for SLE 15 SP4

### PR DESCRIPTION
**Pull request by cjschroder**
update release number entities for SLE 15 SP4

### PR creator: Which product versions do the changes apply to?

When opening a PR, check all versions of the documentation that your PR applies to.

- SLE 15/openSUSE Leap 15.x
  - [x] SLE 15 SP4/openSUSE Leap 15.4 *(current `main`, no backport necessary)*
  - [ ] SLE 15 SP3/openSUSE Leap 15.3
  - [ ] SLE 15 SP2/openSUSE Leap 15.2
  - [ ] SLE 15 SP1
  - [ ] SLE 15 GA
- SLE 12
  - [ ] SLE 12 SP5
  - [ ] SLE 12 SP4
  - [ ] SLE 12 SP3

### PR reviewer only: Have all backports been applied?

The doc team member merging your PR will take care of backporting to older documents.
When opening a PR, do *not* set the following check box.

- [ ] all necessary backports are done

### CODE REVIEW

File: xml/entity-decl.ent [hunk 1]
```diff
@@ -197,9 +197,9 @@
 <!ENTITY productnamereg "<phrase xmlns='http://docbook.org/ns/docbook' role='productname'><phrase os='osuse'>&opensusereg; Leap</phrase><phrase os='sles'>&slsreg;</phrase><phrase os='sled'>&sledreg;</phrase><phrase os='slemicro'>&slemreg;</phrase></phrase>">
 <!ENTITY productnameshort "<phrase xmlns='http://docbook.org/ns/docbook' role='productname'><phrase os='osuse'>&opensuse; Leap</phrase><phrase os='sles'>&slsa;</phrase><phrase os='sled'>&sleda;</phrase><phrase os='slemicro'>&slema;</phrase></phrase>">
 <!ENTITY product-ga    "15">
-<!ENTITY product-sp    "3">
+<!ENTITY product-sp    "4">
 <!ENTITY productnumbershort "&product-ga;.&product-sp;">
```
Comment (sknorr) [inline xml/entity-decl.ent:201]: This is an entirely very ungood entity name. It's not the official short variant of the product version. Instead, it is only a URL component for some SLE registration and Leap repo URLs. To make use cases clearer, I'd suggest replacing with two distinct entities:

```suggestion
<!ENTITY productnumber-regurl "&product-ga;.&product-sp;">
<!ENTITY productnumber-leaprepo "&product-ga;.&product-sp;">
```

There will be fallout from changing this entity name, which must be fixed separately:

```make
[replace with productnumber-regurl]   xml/containers-docker-installation.xml:74:<screen>&prompt.sudo;SUSEConnect -p sle-module-containers/&productnumbershort;/x86_64</screen>
[replace with productnumber-regurl]   xml/klp.xml:232:Activate with: SUSEConnect -p sle-module-live-patching/&productnumbershort;/x86_64 \
[replace with productnumber-regurl]   xml/klp.xml:242:<screen>SUSEConnect -p sle-module-live-patching/&productnumbershort;/x86_64 \
[replace with productnumber-leaprepo] xml/rmt_migrate_from_smt.xml:816:<screen>http://download.opensuse.org/debug/distribution/leap/&productnumbershort;/repo/oss </screen>
[replace with productnumber-leaprepo] xml/rmt_migrate_from_smt.xml:820:<screen>/usr/share/rmt/public/repo/debug/distribution/leap/&productnumbershort;/repo/oss</screen>
[replace with productnumber]          xml/security_ldap.xml:15:    management, and address management. In &productname; &productnumbershort; the
[replace with productnumber]          xml/security_xca.xml:19:        prefer a graphical tool, &sle; &productnumbershort; includes XCA,
```

File: xml/entity-decl.ent [hunk 2]
```diff
@@ -197,9 +197,9 @@
...
-<!ENTITY productnumber "<phrase xmlns='http://docbook.org/ns/docbook' role='productnumber'><phrase os='osuse'>15.3</phrase><phrase os='sles;sled'>&product-ga; SP&product-sp;</phrase><phrase os='slemicro'>5.1</phrase></phrase>">
+<!ENTITY productnumber "<phrase xmlns='http://docbook.org/ns/docbook' role='productnumber'><phrase os='osuse'>15.4</phrase><phrase os='sles;sled'>&product-ga; SP&product-sp;</phrase><phrase os='slemicro'>5.1</phrase></phrase>">
```
Comment (sknorr) [inline xml/entity-decl.ent:202]: It's probably increment weather in Microtown too:

```suggestion
<!ENTITY productnumber "<phrase xmlns='http://docbook.org/ns/docbook' role='productnumber'><phrase os='osuse'>15.4</phrase><phrase os='sles;sled'>&product-ga; SP&product-sp;</phrase><phrase os='slemicro'>5.2</phrase></phrase>">
```

**Review by sknorr — CHANGES_REQUESTED** [review]
* 1000!
* Not sure if this is cleared with Tanja, if not, please do so.
* Check whether you can update the `dm:` Bugzilla things in MAIN.SLEDS.xml and the article files that contain it already. (Make sure that the requisite products already exist in Bugzilla first.)
* Update the version numbers in `.github/PULL_REQUEST_TEMPLATE.md`.

File: xml/entity-decl.ent (hunk 1 above)
Comment (cjschroder) [reply xml/entity-decl.ent:201]: @taroth21 asked that SLES and SLED be changed, I expect Jana is responsible for SLE Micro. See doku-intern 'SLE 15 SP4: Updating the needed infrastructure Wed, 3 Nov 2021'

Comment (cjschroder) [reply xml/entity-decl.ent:201]: I agree that your suggested entity name is an improvement-- shouldn't all SLE 15 releases have this change?

File: xml/entity-decl.ent (hunk 2 above)
Comment (cjschroder) [reply xml/entity-decl.ent:202]: ahaha increment weather :suspect:

File: xml/entity-decl.ent (hunk 1 above)
Comment (cjschroder) [reply xml/entity-decl.ent:201]: see https://github.com/SUSE/doc-sle/pull/1001 for MAIN file changes

Comment (cjschroder) [reply xml/entity-decl.ent:201]: eh too much overthinking re sle micro...

**Comment by sknorr** [comment]
Right, merged that with changes (since you hadn't fixed the build breakage after the entity name change).

**Comment by sknorr** [comment]
I've also propagated the entity name changes across 15 SPx branches: 15 SP3 e2c0488 / 15 SP2 4b8259e / 15 SP1 1f0f3e2 / 15 GA 4481c57.

**Comment by cjschroder** [comment]
> Right, merged that with changes (since you hadn't fixed the build breakage after the entity name change).

 Are you in a hurry? I wasn't finished :)

**Pull request merged**